### PR TITLE
Feature 6 matching system

### PR DESCRIPTION
**Pull request by MacroSony**
Updated Matcher so that it return randomly. Updated Tests for Matcher and MatchManager. Updated comments for all files.

### CODE REVIEW

File: src/main/java/matching_system/MatchManager.java [hunk 1]
```diff
@@ -26,10 +28,12 @@ public MatchManager(MatchOutputBoundary matchOutputBoundary, MatchRequestModel m
             userAccountArrayList.add(rm.getUserAccount());
         }
         this.match = new Matcher(user,userAccountArrayList);
+        create(matchRequestModel);
```
Comment (cyltaracai) [inline src/main/java/matching_system/MatchManager.java:31]: This method returns MatchResponseModel, right? But the result is not saved? What's the purpose of calling this method here?

File: src/main/java/matching_system/MatcherUITemplateJames.java [hunk 2]
```diff
@@ -17,14 +17,19 @@ public class MatcherUITemplateJames extends JPanel implements ActionListener {
...
 
+    /** A preset of UI panel that display 5 user info which are included in the matchResponseModel
+     * @param matcherUIController UIController to response to actions
+     * @param requesterEmail email of the requester
+     * @param matchUIPresenter the presenter that stores matchedRespondModel
+     * */
     public MatcherUITemplateJames(MatcherUIController matcherUIController,
                              String requesterEmail,
-                             MatchResponseModel matchResponseModel){
+                             MatchOutputBoundary matchUIPresenter){
```
Comment (cyltaracai) [inline src/main/java/matching_system/MatcherUITemplateJames.java:27]: MatcherUITemplateJames cannot call MatchOutpurBoundary directly, but can call MatcherUIPresenter. So change the type of matchUIPresenter to MatchUIPresenter.

**Review by cyltaracai — CHANGES_REQUESTED** [review]
1. Identified one violation of clean architecture in MatcherUITemplateJames. Please see the detailed comment. 
2. The MatchManager constructor appears to me that it violates the single responsibility principle. It also calls a method but does not save the returned results, which seems a bit weird to me.

File: src/main/java/matching_system/MatchManager.java (hunk 1 above)
Comment (MacroSony) [reply src/main/java/matching_system/MatchManager.java:31]: This function actually also saves the MatchResponseModel to the Presenter, so here I called it to put the MatchResponseModel into the presenter.

File: src/main/java/matching_system/MatcherUITemplateJames.java (hunk 2 above)
Comment (MacroSony) [reply src/main/java/matching_system/MatcherUITemplateJames.java:27]: Changed

**Review by cyltaracai — DISMISSED** [review]
The java docs/comments for test classes need to be more specific. For example, brief discussion about test scenarios.

**Review by RitatataVan — DISMISSED** [review]
Great work!

**Review by hzhqwq — DISMISSED** [review]

**Review by cyltaracai — APPROVED** [review]
approve to merge

**Pull request merged**